src/style/style.js

```diff
@@ -568,7 +568,12 @@ class Style extends Evented {
 
         layer.setEventedParent(this, {layer: {id: id}});
 
-        const index = before ? this._order.indexOf(before) : this._order.length;
+
+        const beforeIndex = before ? this._order.indexOf(before) : -1;
```

Comment:
`this._order.length` can never be `-1`, so the conditionals can be simplified:

```
const index = before ? this._order.indexOf(before) : this._order.length;
if (index === -1) {
    // error
}

this._order.splice(index, ...);
```

Reply:
ah good point! I still had to use `if (before && index === -1)` for flow to allow me to  include `before` in the error message.